c/recovery.c

```diff
@@ -865,27 +971,39 @@ int recoveryEstablishRouter(int flags) {
       :
   );
 
-  /* ESTAEX */
-  char estaexFlags = 0;
-  if (flags & RCVR_ROUTER_FLAG_NON_INTERRUPTIBLE) {
-    estaexFlags |= RCVR_ESTAEX_FLAG_NON_INTERRUPTIBLE;
-  }
-  if (flags & RCVR_ROUTER_FLAG_RUN_ON_TERM) {
-    estaexFlags |= RCVR_ESTAEX_FLAG_RUN_ON_TERM;
-  }
 
-  ESTAEXFeedback feedback = setESTAEX(getRecoveryRouterAddress(), context, estaexFlags);
-  if (feedback.returnCode != 0) {
+  if (isTCB) {
+
+    /* ESTAEX */
+    char estaexFlags = 0;
+    if (flags & RCVR_ROUTER_FLAG_NON_INTERRUPTIBLE) {
+      estaexFlags |= RCVR_ESTAEX_FLAG_NON_INTERRUPTIBLE;
+    }
+    if (flags & RCVR_ROUTER_FLAG_RUN_ON_TERM) {
+      estaexFlags |= RCVR_ESTAEX_FLAG_RUN_ON_TERM;
+    }
+
+    ESTAEXFeedback feedback = setESTAEX(getRecoveryRouterAddress(), context,
+                                        estaexFlags);
+    if (feedback.returnCode != 0) {
 #if RECOVERY_TRACING
-    printf("error: set ESTAEX RC = %d, RSN = %d\n", feedback.returnCode, feedback.reasonCode);
+      printf("error: set ESTAEX RC = %d, RSN = %d\n",
+             feedback.returnCode, feedback.reasonCode);
 #endif
-    if (previousESPIEToken != 0) {
-      resetESPIE(previousESPIEToken);
-      previousESPIEToken = 0;
+      if (previousESPIEToken != 0) {
+        resetESPIE(previousESPIEToken);
+        previousESPIEToken = 0;
+      }
+      storageRelease((char *)context, sizeof(RecoveryContext));
```

Comment:
What is the purpose of the cast?

Reply:
Can't tell. This could have been copied/movied as is (with the cast) from another function in the original code.